### PR TITLE
Add support to gitClientFactory for CloneURI Remote Resolver

### PR DESCRIPTION
**Pull request by mpherman2**
Starting the work to add gerrit support for InRepo Config. 
/assign @cjwagner @chaodaiG

### CODE REVIEW

File: prow/git/v2/remote.go
Comment (stevekuznetsov) [inline prow/git/v2/remote.go:164]: nit:
```suggestion
		return "", errors.New("not implemented")
```

Comment (stevekuznetsov) [reply prow/git/v2/remote.go:164]: IMO could also be a panic

Comment (stevekuznetsov) [reply prow/git/v2/remote.go:164]: Ideally, there would be some way to tell before merging a PR that a gerrit client is calling this and should not be

File: prow/git/v2/remote.go
Comment (stevekuznetsov) [inline prow/git/v2/remote.go:151]: Why hold the URL, not the string?

Comment (mpherman2) [reply prow/git/v2/remote.go]: Good point. Changed

File: prow/git/v2/remote.go
Comment (mpherman2) [reply prow/git/v2/remote.go]: When I implement the gerrit support I will keep this in mind as something to add. I added a TODO

File: prow/git/v2/client_factory.go
Comment (chaodaiG) [inline prow/git/v2/client_factory.go:139]: Would expect something more strict than `CloneURI != nil -> Using cloneURIResolverFactory`, just like `o.UseSSH` below.

File: prow/git/v2/remote.go
Comment (mpherman2) [reply prow/git/v2/remote.go]: Decided to actually have this return CloneURI as well. Github clients can use CloneURIRemoteResolverFactory when CentralRemote and PublisherRemote are both CloneURI. No need to leave this for gerrit only.

File: prow/git/v2/client_factory.go
Comment (mpherman2) [reply prow/git/v2/client_factory.go]: Discussed offline. Added support for PublisherRemote for CloneURIResolverFactory so adding CloneURI won't cause surprising behavior

**Review by chaodaiG — APPROVED** [review]
/hold

**Pull request merged**
